CMakeLists.txt

```diff
@@ -164,6 +165,24 @@ if(NOT mxnet_LINKER_LIBS)
   set(mxnet_LINKER_LIBS "")
 endif(NOT mxnet_LINKER_LIBS)
 
+if(USE_NGRAPH)
+  include(cmake/ngraph.cmake)
+  if(NOT MSVC)
+    set(ARCH_OPT_FLAGS "-mtune=generic")
+  endif()
+  
+  include_directories(${CMAKE_BINARY_DIR}/3rdparty/ngraph/include)
+  include_directories($src/ngraph)
+  
+  add_definitions(-DUSE_MKL=1)
+  add_definitions(-DCUB_MKL=1)
+  add_definitions(-DMXNET_USE_NGRAPH=1)
+  list(APPEND mxnet_LINKER_LIBS ngraph_interface_lib)
+  if(USE_NGRAPH_DISTRIBUTED)
+    list(APPEND mxnet_LINKER_LIBS ${MPI_CXX_LIBRARIES})
```

Comment:
@sfvaroglu I'm trying to link MPI here, but I haven't actually tested this yet.

Reply:
I see. Let me know if this PR resolves the MPI_Init issue.